src/main/scala/io/citrine/lolo/trees/Nodes.scala

```diff
@@ -181,38 +177,20 @@ class InternalModelNode[T <: PredictionResult[Any]](
   override def getTrainingWeight(): Double = trainingWeight
 }
 
-/**
-  * Average the training data to make a leaf prediction
-  *
-  * @param trainingData to train on
-  */
-class TrainingLeaf[T](
-    trainingData: Seq[(Vector[AnyVal], T, Double)],
-    leafLearner: Learner,
-    depth: Int,
-    rng: Random
-) extends TrainingNode(
-      trainingData = trainingData,
-      remainingDepth = 0
-    ) {
+/** A leaf defined by a training result. */
+trait TrainingLeaf[T] extends TrainingNode[T] {
```

Comment:
Previously there was this concrete `TrainingLeaf` class that was being used by the classification tree, and also a `RegressionTrainingLeaf` class that was very similar but specialized for regression.
I think that this high-level `Node` file should be limited to traits/abstract classes or implementations that are truly universal. So I redefined this as a trait and implemented a `ClassificationTrainingLeaf`. In practice, the two training leaves are identical except for their feature importance calculation.